neo4j/internal/bolt/bolt3_test.go

```diff
@@ -512,4 +513,39 @@ func TestBolt3(ot *testing.T) {
 		AssertNil(t, sum)
 		AssertError(t, err)
 	})
+
+	outer.Run("Updates idle date on every new message", func(t *testing.T) {
+		ctx := context.Background()
+		testStart := time.Now()
+		bolt, cleanup := connectToServer(t, func(srv *bolt3server) {
+			srv.accept(3)
+			srv.waitForReset()
+			srv.sendSuccess(map[string]interface{}{})
+		})
+		defer cleanup()
+		defer bolt.Close(ctx)
+
+		firstIdleDate := bolt.IdleDate()
+		AssertAfter(t, firstIdleDate, testStart)
+
+		bolt.ForceReset(ctx)
+		AssertAfter(t, bolt.IdleDate(), firstIdleDate)
+	})
+
+	outer.Run("Does not update idle date on error", func(t *testing.T) {
```

Comment:
```suggestion
	outer.Run("Does not update idle date on disconnect", func(t *testing.T) {
```
I'd expect it to update the idle date on receiving `FAILURE`.

Reply:
since this is a real TCP server under the hood, it's kinda hard to produce anything else than a disconnect error.
However, I parameterized *when* the error happens, by including those cases into the... matrix.